src/circuit/gadget/ecc/chip/add.rs

```diff
@@ -407,22 +407,50 @@ pub mod tests {
         assert_ne!(p_val, q_val);
 
         // Check complete addition P + (-P)
-        p.add(layouter.namespace(|| "P + (-P)"), p_neg)?;
+        {
+            let sum = p.add(layouter.namespace(|| "P + (-P)"), p_neg)?;
+            assert_eq!(&sum, zero);
```

Comment:
This isn't what I was thinking of when I said "constrain": I mean that we add a circuit constraint that `sum == zero`. This is probably fine for testing purposes, but I worry that with the `Point` gadget implementing `PartialEq`, someone else might write code like `if sum == zero { constrain something }` which results in a non-deterministic circuit.